### DISCUSSION TITLE
Dynamic Entity Value Recognition

### DISCUSSION THREAD

**Discussion by ryan-loughner** [speechly/speechly#215] [Q&A] [answered]
Hello, quick question regarding the capabilities of Speechly. I want to run the following voice command 
`show client abc(client)` where "abc" can be any client. For example, I could gain 20 clients in a day, and I want to begin using those new companies in the voice commands. Does Speechly support a dynamic entity input or do the client names need to be pre-populated in a list? I'd have to redeploy when I gain new clients, unless there's a better solution.

**Comment by thisismatu** [comment, marked as answer]
Hi @ryan-loughner  👋 

Our entity detection system is based on machine learning, and can often generalize to entity values that are not present in the configuration. However, entity values that explicitly appear in the training data are more likely to be correctly recognized.

For the best results you should have:

**Enough entity training data**

In your case this would be the client list. In the beginning i would update more often but as the system learns to generalize the updates would become more infrequent. For changes to take effect, a deployment is required.

**Variance intent expression**

Some different ways of expressing the "show client" intent.

```
*show show me $clients(client)
*show show $clients(client)
*show give me $clients(client)
*show give $clients(client)
etc.
```
